src/bgw/job_stat.c

```diff
@@ -203,9 +203,9 @@ calculate_next_start_on_failure(TimestampTz finish_time, int consecutive_failure
 	float8 multiplier = (consecutive_failures > MAX_FAILURES_MULTIPLIER ? MAX_FAILURES_MULTIPLIER :
 																		  consecutive_failures);
 	MemoryContext oldctx;
-	Assert(consecutive_failures > 0);
-	int64 max_slots =
-		(1 << consecutive_failures) - 1; /* 2^(consecutive_failures) - 1, at most 2^20 */
+	Assert(consecutive_failures > 0 && consecutive_failures < 63);
+	/* 2^(consecutive_failures) - 1, at most 2^20 */
+	int64 max_slots = (((int64) 1) << consecutive_failures) - 1;
```

Comment:
```suggestion
	int64 max_slots = (INT64CONST(1) << consecutive_failures) - INT64CONST(1);
```
for consistency with `bit_array_low_bits_mask()` and PG code.

Reply:
Seems pointless to have a macro for this, but if that is the way they roll, I'll do it as well. Changed.